spec/unit/system_description_store_spec.rb

```diff
@@ -228,4 +228,46 @@
       }.to raise_error(Machinery::Errors::SystemDescriptionError, /#{new_name}/)
     end
   end
+
+  describe "#swap" do
+    let(:store) { SystemDescriptionStore.new(test_base_path) }
+    let(:test_name_2) { "description2" }
+
+    before(:each) do
+      create_machinery_dir
+
+      FileUtils.mkdir_p(File.join(test_base_path, test_name_2))
+      File.write(File.join(test_base_path, test_name_2, "manifest.json"), test_manifest)
+    end
+
+    it "swaps two existing SystemDescriptions" do
+      FileUtils.touch(File.join(test_base_path, test_name_2, "test.txt"))
+
+      expect(
+        Dir.entries(File.join(test_base_path, test_name))
+      ).to match_array([".", "..", "manifest.json"])
+      expect(
+        Dir.entries(File.join(test_base_path, test_name_2))
+      ).to match_array([".", "..", "manifest.json", "test.txt"])
```

Comment:
It would be good to also check that the content of the `manifest.json` is correctly swapped to have a full check.